### PR TITLE
Register & Login Form

### PR DESCRIPTION
**Pull request by renzowolox**
## Summary
* Register form
* Login form
* Tests

## Screenshots
https://user-images.githubusercontent.com/53195970/129242902-9f697043-1b42-4480-842b-6c91e1679fe8.mov
https://user-images.githubusercontent.com/53195970/129941358-290de904-1c49-481a-9ad6-c1a75c59527e.mov

## Trello cards
* https://trello.com/c/Vrx2U4cb/21-front-registration-form
* https://trello.com/c/AaZu6eME/34-front-login-form
* https://trello.com/c/UXiQhGy8/33-front-fix-tests

### CODE REVIEW

File: src/screens/Authentication/Login/index.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    setErrors(undefined);
+    mutate(params);
+  };
+
+  return (
+    <Layout>
+      <div className={`full-width column center ${styles.container}`}>
+        <h1 className={`m-bottom-2 ${styles.title}`}>{t('signIn')}</h1>
+        <Link to="/register" className={styles.linkToAccount}>
```
Comment (hernan-clich) [inline src/screens/Authentication/Login/index.tsx:57]: Please import the `PATHS` and try something like: `<Link to={PATHS.register} ...>` so if the route ever changes you only need to change it in one single place.

File: src/screens/Authentication/Register/index.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    setErrors(undefined);
+    mutate(params);
+  };
+
+  return (
+    <Layout>
+      <div className={`full-width column center ${styles.container}`}>
+        <h1 className={`m-bottom-2 ${styles.title}`}>{t('signUp')}</h1>
+        <Link to="/login" className={styles.linkToAccount}>
```
Comment (hernan-clich) [inline src/screens/Authentication/Register/index.tsx:57]: Same here, use `PATHS` constant

**Review by hernan-clich — COMMENTED** [review]
I noticed a few things, it'd be great if you could take a look.

File: src/components/FormInput/index.tsx [hunk 3]
```diff
@@ -45,7 +47,9 @@ function FormInput({
 }: Props) {
   const InputComponent = isTextarea ? 'textarea' : 'input';
   const showError =
-    (touched === undefined || touched) && error && (submitCount === undefined || submitCount > 0);
+    (touched === undefined || touched) &&
```
[the remark this answers is not in the document]
Comment (renzowolox) [reply src/components/FormInput/index.tsx:50]: I understand that yes, this component comes with bootstrap. And I just added the condition showErrorWithoutText.

File: src/components/UserForm/index.tsx [hunk 4]
```diff
@@ -1,17 +1,25 @@
+/* eslint-disable complexity */
```
[the remark this answers is not in the document]
Comment (renzowolox) [reply src/components/UserForm/index.tsx:1]: This comment shouldn't be here, I forgot to remove it.

**Review by PipeWolox — APPROVED** [review]

File: src/components/Routes/index.tsx [hunk 5]
```diff
@@ -22,11 +22,13 @@ function Routes() {
...
         } = data;
-        userDispatch(
-          authActions.setUser({
-            user
-          })
-        );
+        if (user) {
+          userDispatch(
+            authActions.setUser({
+              ...user
```
Comment (PipeWolox) [inline src/components/Routes/index.tsx:28]: You could just pass the user right?
`authActions.setUser(user)`

File: src/screens/Authentication/Login/index.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  const { mutate, isLoading } = useMutation((params: { user: UserFormKeys }) => login(params), {
+    onSettled: (data) => {
+      if (data?.data && data?.ok) {
+        const {
+          data: { user }
+        } = data;
+        if (user) {
+          userDispatch(
+            authActions.setUser({
+              ...user
```
Comment (PipeWolox) [inline src/screens/Authentication/Login/index.tsx:33]: You could just pass the user right?

`authActions.setUser(user)`

File: src/screens/Authentication/Register/index.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  const { mutate, isLoading } = useMutation((params: { user: UserFormKeys }) => signup(params), {
+    onSettled: (data) => {
+      if (data?.data && data?.ok) {
+        const {
+          data: { user }
+        } = data;
+        if (user) {
+          userDispatch(
+            authActions.setUser({
+              ...user
```
Comment (PipeWolox) [inline src/screens/Authentication/Register/index.tsx:33]: equal to te previous

**Review by PipeWolox — CHANGES_REQUESTED** [review]

**Review by hernan-clich — COMMENTED** [review]
Once the comments from Pipe are resolved I'll give you my approval!

**Review by PipeWolox — APPROVED** [review]
Great!! Good Work!! 👍

**Review by hernan-clich — APPROVED** [review]
Nice job man! 💯

**Pull request merged**